corehq/apps/integration/models.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import absolute_import
+
+from memoized import memoized
+from corehq import toggles
+
+from django.db import models
+
+
+class SimprintsIntegration(models.Model):
+    domain = models.CharField(max_length=128, unique=True)
+    is_enabled = models.BooleanField(default=False)
+    project_id = models.CharField(max_length=255)
+    user_id = models.CharField(max_length=255)
+    module_id = models.CharField(max_length=255)
+
+
+class ApplicationIntegrationMixin(object):
+    """
+    Contain all integration options in one place for Application object.
+    assumes access to self.domain from Application
+    """
+
+    @property
+    @memoized
+    def is_biometric_enabled(self):
+        existing, _ = SimprintsIntegration.objects.get_or_create(
```

Comment:
Why `get_or_create`? Seems we might end up with a bunch of harmless but also useless `SimprintsIntegration` objects. Also, I wouldn't expect an is_enabled-type method to have side effects.

Reply:
> Seems we might end up with a bunch of harmless but also useless `SimprintsIntegration` objects. Also, I wouldn't expect an is_enabled-type method to have side effects.

It would only save if they save the form, otherwise the object is created and then discarded.

Reply:
aha, thanks for explaining